index.js

```diff
@@ -34,6 +34,17 @@ module.exports = function (obj, opts) {
 			var key = keys[i];
 			var val = x[key];
 
+			if (deep && Array.isArray(val)) {
+				var retArr = [];
+
+				for (var j = 0; j < val.length; j++) {
```

Comment:
Yeah, I tried keeping it as close as possible to the current code style wise. [This](https://github.com/sindresorhus/sort-keys/blob/f977719f258f4e117be7821e3ed70e7c99468f6e/index.js#L33-L52) whole part can be rewritten.